pkgs/applications/graphics/glabels-qt/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, fetchFromGitHub
+, cmake
+, mkDerivation
+, qttools
+}:
+
+mkDerivation {
+  pname = "glabels-qt";
+  version = "unstable-2021-02-06";
+
+  src = fetchFromGitHub {
+    owner = "jimevins";
+    repo = "glabels-qt";
+    rev = "glabels-3.99-master564";
+    hash = "sha256-SdNOkjspqf6NuuIBZDsJneY6PNrIiP4HU46wDpBLt9Y=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    qttools
+  ];
+
+  meta = with lib; {
+    description = "GLabels Label Designer (Qt/C++)";
+    homepage = "https://github.com/jimevins/glabels-qt";
+    license = licenses.gpl3Only;
+    maintainers = [ maintainers.matthewcroughan ];
+    platforms = lib.platforms.unix;
```

Comment:
```suggestion
    platforms = lib.platforms.unix;
    # TODO
    broken = stdenv.isDarwin;
```
not sure what to write into the comment

```
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:45:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/type_traits:417:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/cstddef:37:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:45:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/type_traits:417:
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/cstddef:49:9: error: no member named 'ptrdiff_t' in the global namespace
using ::ptrdiff_t;
      ~~^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:45:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/type_traits:418:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:47:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/utility:205:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:668:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/typeinfo:60:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/exception:83:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:671:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/new:93:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:673:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/limits:121:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:674:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:423:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:674:
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:562:13: error: unknown type name 'ptrdiff_t'
    typedef ptrdiff_t difference_type;
            ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:623:56: error: unknown type name 'ptrdiff_t'
template<class _Category, class _Tp, class _Distance = ptrdiff_t,
                                                       ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:989:66: error: unknown type name 'ptrdiff_t'
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                                                                 ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:989:54: error: template parameter missing a default argument
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                                                     ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/iterator:989:27: note: previous default template argument defined here
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                          ^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:677:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/tuple:154:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:681:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:571:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/__threading_support:14:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/chrono:831:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:681:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:576:
/tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/build/model/../version:1:1: error: expected unqualified-id
3.99-Unkonwn?
^
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.cpp:21:
In file included from /tmp/nix-build-glabels-qt-unstable-2021-02-06.drv-0/source/model/Category.h:25:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/QString:1:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qstring.h:49:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qchar.h:43:
In file included from /nix/store/zm1hlkvpqwg22bkdmny0zd0by4ksb2s9-qtbase-5.15.9-dev/include/QtCore/qglobal.h:142:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/algorithm:643:
In file included from /nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/memory:681:
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:1047:73: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_add(__cxx_atomic_base_impl<_Tp*> volatile* __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                        ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:1052:65: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_add(__cxx_atomic_base_impl<_Tp*> * __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:1068:73: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> volatile* __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                        ^
/nix/store/mhvjwl82jxqbfqkk6lc652wm2bbiimqg-libcxx-11.1.0-dev/include/c++/v1/atomic:1073:65: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> * __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [model/CMakeFiles/Model.dir/build.make:124: model/CMakeFiles/Model.dir/Category.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:397: model/CMakeFiles/Model.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
error: builder for '/nix/store/dk94ksqzqp92wvq1n2p0x2cn742p56x6-glabels-qt-unstable-2021-02-06.drv' failed with exit code 2;
```

Reply:
@SuperSandro2000  It's probably not worth commenting, this software was probably never made to work on a Mac.

Reply:
Then we should set meta.platforms to linux and with TODO I meant that a fitting comment should be added

Reply:
@SuperSandro2000 I've set the platform to Linux, but I don't intend to spend any time getting this working on a Mac, and I'm not sure who else would, so I'm not sure why we should add a TODO.

Reply:
That's fine, I just wanted you to replace the TODO with an actual comment so that people at least have an idea why this failed.